darshan-util/pydarshan/tests/test_data_access_by_filesystem.py

```diff
@@ -0,0 +1,272 @@
+                                                                                                 processing_func=processing_func,
+                                                                                                 mod=mod,
+                                                                                                 verbose=verbose)
+        assert_series_equal(actual_read_groups, expected_read_groups)
+        assert_series_equal(actual_write_groups, expected_write_groups)
+    else:
+        with pytest.raises(NotImplementedError):
+            data_access_by_filesystem.unique_fs_rw_counter(report=report,
+                                                           filesystem_roots=filesystem_roots,
+                                                           file_id_dict=file_id_dict,
+                                                           processing_func=processing_func,
+                                                           mod=mod,
+                                                           verbose=verbose)
+
+
+@pytest.mark.parametrize("""file_rd_series,
+                            file_wr_series,
+                            bytes_rd_series,
+                            bytes_wr_series,
+                            filesystem_roots
+                         """, [
+
+     (pd.Series([3.0], index=pd.Index(['/p'], name='filesystem_root'), name='filepath'),
+      pd.Series([14.0], index=pd.Index(['/p'], name='filesystem_root'), name='filepath'),
+      pd.Series([2.145206e+09], index=pd.Index(['/p'], name='filesystem_root'), name='POSIX_BYTES_READ'),
+      pd.Series([1.010878e+12], index=pd.Index(['/p'], name='filesystem_root'), name='POSIX_BYTES_WRITTEN'),
+      ['/p'],
+         ),
+       ])
+def test_plot_data(file_rd_series, file_wr_series, bytes_rd_series, bytes_wr_series, filesystem_roots):
+    # test a few basic properties of the main plotting function
+    fig = plt.figure()
+    data_access_by_filesystem.plot_data(fig=fig,
+                                        file_rd_series=file_rd_series,
+                                        file_wr_series=file_wr_series,
+                                        bytes_rd_series=bytes_rd_series,
+                                        bytes_wr_series=bytes_wr_series,
+                                        filesystem_roots=filesystem_roots)
+    axes = fig.gca()
+    children = axes.get_children()
+    actual_list_text_in_fig = []
+
+    # accumulate text added via ax.text()
+    # by the function
+    for child in children:
+        if isinstance(child, matplotlib.text.Text):
+            actual_list_text_in_fig.append(child.get_text())
+
+    for expected_text_entry in [matplotlib.text.Text(0, 1, '# files read'),
+                                matplotlib.text.Text(0, 0, '# files written')]:
+        assert expected_text_entry.get_text() in actual_list_text_in_fig
```

Comment:
Here's an example of testing some simple properties of the plot--initially for text that we expect to always be present.

Presumably it is also possible to pull out the position of that text, and of most other plot elements. While we probably don't want to lock this down too much, one thing I'd eventually like to try to do is reproduce one of the problems observed during code review above and then get the test to fail because of that particular i.e., positioning issue, etc.

Reply:
Using i.e., `dir()` on various `matplotlib` objects can give a good sense for their various attributes/handles we can use for some select regression tests maybe.